src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/AppBase.cs

```diff
@@ -378,12 +379,7 @@ private async Task<DataElement> StorePDF(Stream pdfStream, Instance instance, Te
 
         private string GetValidFileName(string fileName)
         {
-            foreach (char c in System.IO.Path.GetInvalidFileNameChars())
-            {
-                fileName = fileName.Replace(c, '_');
-            }
-
-            fileName = Uri.EscapeDataString(fileName);
+            fileName = Uri.EscapeDataString(fileName.AsFileName(false));
```

Comment:
Does this also cover spaces in the file name?

Reply:
Space is a valid character in most cases. I have not taken any special care related to spaces because there were none in the original code.